### ISSUE TITLE
A bug when I added encoder layers 

### ISSUE THREAD

**Issue by RuifMaxx** [oliverguhr/transformer-time-series-prediction#13]
Thanks very much for your code. However, there are some difference between from the tutorials of PyTorch:  [SEQUENCE-TO-SEQUENCE MODELING WITH NN.TRANSFORMER AND TORCHTEXT](https://pytorch.org/tutorials/beginner/transformer_tutorial.html)

According to [https://www.zhihu.com/question/67209417/answer/1264503855](https://www.zhihu.com/question/67209417/answer/1264503855), with the addition of `self. `This leads to `self. Encoder_ Layers` are counted as parameters of module, but only `self. Transformer`is used in network operation_ The nlayers copied from encoder are the parameters in `nn.transformerencoderlayer` That is to say, `self. Encoder_ Layers` do not participate in model operation, so there is no gradient in backward, which leads to training errors.

**Comment by oliverguhr** [comment]
Well thank you! I did not think that torch would use this layer as long as I am not referencing it.

**Comment by RuifMaxx** [comment]
Well thanks for you reply
